src/App.tsx

```diff
@@ -14,6 +36,132 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const getTodosFromApi = useCallback(async (userId: number) => {
+    try {
+      const response = await getTodos(userId);
+
+      setTodos(response);
+      setTempTodo(null);
+      setIsEditing(null);
+      setIsAllSelected(response.every(todo => !todo.completed));
+    } catch (error) {
+      setErrorMessage('Unable to load todos');
+      setIsHidden(false);
+    }
+  }, []);
+
+  const createTodo = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!inputTitle) {
+      setErrorMessage('Title can\'t be empty');
+      setIsHidden(false);
+
+      return;
+    }
+
+    if (user) {
+      const newTodo = {
+        id: 0,
+        userId: user.id,
+        title: inputTitle,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+
+      try {
+        await createTodos(newTodo);
+        getTodosFromApi(user.id);
+      } catch {
+        setErrorMessage('Unable to add a todo');
+        setIsHidden(false);
+        setTempTodo(null);
+      }
+
+      setInputTitle('');
+    }
+  };
+
+  const setTodoSelected = useCallback(async (todo: Todo) => {
+    try {
+      await updateTodo(todo, !todo.completed);
+
+      if (user) {
+        getTodosFromApi(user.id);
+      }
+    } catch {
+      setErrorMessage('Unable to update a todo');
+      setIsHidden(false);
+    }
+  }, []);
+
+  const selectAll = () => {
+    const todoFunctions = todos.map(async (todo) => {
+      if ((!todo.completed && isAllSelected)
+        || (todo.completed && !isAllSelected)) {
+        try {
+          await updateTodo(todo, !todo.completed);
```

Comment:
Don't need to wait for update if you then call Promise.all(). And why do you use array map if your callback doesn't return anything?
First, fill array with promises (return it here, not await). Then your Promise.all() will work correctly.